sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -8,10 +8,114 @@
 
 from collections import defaultdict
 
+from datadog_checks.base import AgentCheck, ConfigurationError
+
 # Queries
 ALL_INSTANCES = 'ALL'
 
 
+class CustomQueryMetric(object):
```

Comment:
Any reason we can't use the QueryManager?

Reply:
No reason we couldn't substitute it, but since these queries are a special case rather than the base part of the check, thought it would be better to code it directly.

Reply:
We lose all these though https://datadoghq.dev/integrations-core/base/databases/#transformers

Reply:
In the cases where that happens we need to special handle it because of windows performance counters, and run it through a translator like this: https://github.com/DataDog/integrations-core/blob/42b4d07b1223f6fd4af5d896bbd84a9e0b1efd3d/sqlserver/datadog_checks/sqlserver/sqlserver.py#L354

Those types of metrics are currently handled through [custom metrics](https://www.datadoghq.com/blog/sql-server-metrics/#custom-datadog-metrics-with-the-performance-counters-view).

Reply:
Migrated!